src/protocol_nmea.cpp

```diff
@@ -303,217 +418,213 @@ int did_dev_info_to_nmea_info(char a[], const int aSize, dev_info_t &info)
 		info.buildTime[0], info.buildTime[1], info.buildTime[2], info.buildTime[3], // 9
 		info.addInfo);			// 10
 		
-	unsigned int checkSum = ASCII_compute_checksum((uint8_t*)(a+1), n);
-	n += SNPRINTF(a+n, aSize-n, "*%.2x\r\n", checkSum);
-	return n;	
+	return nmea_sprint_footer(a, aSize, n);
 }
 
 int tow_to_nmea_ptow(char a[], const int aSize, double imuTow, double insTow, unsigned int gpsWeek)
 {
-	int n = SNPRINTF(a, aSize, "$PTOW");
-	n += SNPRINTF(a+n, aSize-n, ",%.6lf", imuTow);			// 1
-	n += SNPRINTF(a+n, aSize-n, ",%.6lf", insTow);			// 2
-	n += SNPRINTF(a+n, aSize-n, ",%u", gpsWeek);			// 3
-	
-	unsigned int checkSum = ASCII_compute_checksum((uint8_t*)(a+1), n);
-	n += SNPRINTF(a+n, aSize-n, "*%.2x\r\n", checkSum);
-	return n;	
+	int n = ssnprintf(a, aSize, "$PTOW");
+	nmea_sprint(a, aSize, n, ",%.6lf", imuTow);			// 1
```

Comment:
I like the new use of nmea_sprint() though I'd probably use `nmea_sprintf` to be clear that its a printf derivative.  But, I'm not sure the meaning of the  inline-comments numbering // 1, 2, 3, etc...  Either these should go away, or there should be some supporting comments to communicate their purpose.